### PR TITLE
Fix the preview of text files containing BOM for utf8

### PR DESCRIPTION
**Pull request by jvillafanez**
Adjusted range so the BOM isn't detected as an arabic char.
Latin is also added as part of possible fonts, not just if there isn't
any match

## Description
BOM was detected as arabic char. Alternative fix to https://github.com/owncloud/core/pull/39644

## Related Issue
https://github.com/owncloud/core/issues/39645

## How Has This Been Tested?
1. Create a text file containing the BOM and some latin chars
2. Upload the file to ownCloud
3. The preview should show correctly.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: lib/private/Utf8Analyzer.php [hunk 1]
```diff
@@ -92,7 +92,7 @@ class Utf8Analyzer {
 		['range' => [0xf900, 0xfaff], 'script' => 'Han'],
 		['range' => [0xfb1d, 0xfb4f], 'script' => 'Hebrew'],  // some unicode chars aren't assigned
 		['range' => [0xfb50, 0xfdff], 'script' => 'Arabic'],
-		['range' => [0xfe70, 0xfeff], 'script' => 'Arabic'],
+		['range' => [0xfe70, 0xfefc], 'script' => 'Arabic'],
```
Comment (jvillafanez) [inline lib/private/Utf8Analyzer.php:95]: For reference: https://en.wikipedia.org/wiki/Arabic_Presentation_Forms-B

**Comment by JanAckermann** [comment]
We should check manually if the code now works with all BOMs ```0xEF,0xBB,0xBF```

**Comment by jvillafanez** [comment]
Added unit tests to verify the text analyzer process the BOM correctly.
I've also checked manually that a file with latin chars and a BOM marker at the beginning is previewed properly

**Review by AlexAndBear — APPROVED** [review]

File: changelog/unreleased/39669 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Bugfix: Text previews had faulty content if BOM was present
+
+The BOM was incorrectly detected and was causing ownCloud to choose
+the wrong font for the text, showing unexpected results.
+The BOM is now processed correctly and the preview is show as expected
```
Comment (JanAckermann) [inline changelog/unreleased/39669:5]:
```suggestion
The BOM is now processed correctly and the preview is shown as expected
```

**Pull request merged**